examples/local_helpers_stream.py

```diff
@@ -0,0 +1,62 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+import asyncio
+from asyncio import to_thread
+from typing import AsyncIterator
+
+import numpy as np
+import sounddevice as sd
+import torch
+from context import iftk
+
+from iftk.helpers import pyaudio, silero_vad, transformers, whisper, xtts
+
+MODEL_ID = "TinyLlama/TinyLlama-1.1B-Chat-v1.0"
+CHUNK = 512 * 100
+RATE = 16000
+messages = [{"role": "system", "content": "Respond in a few sentences."}]
+
+
+async def main():
```

Comment:
I think this is OK to land. An interesting exercise for the future would be to unroll these loops. Right now, the computation is still happening in distinct phases, but we could pass streams at each point and flatten the whole thing out.

What it takes to do that is to create Stream-type connector functions to glue the pieces together. For example, to wait for a full sentence, to play audio, etc.